ORK1Kit/ORK1Kit/Common/CEVRK1Theme.m

```diff
@@ -49,177 +77,256 @@ + (instancetype)themeForElement:(id)element {
     }
     if ([element isKindOfClass:[ORK1StepViewController class]]) {                                                  // is stepViewController, jump to task for theme
         id <ORK1Task> task = [(ORK1StepViewController *)element taskViewController].task;
-        return [CEVRK1Theme themeForElement:task];
+        CEVRK1Theme *taskTheme = task.cev_theme;
+        CEVRK1Theme *stepTheme = ((ORK1StepViewController *)element).step.cev_theme;
+        return [CEVRK1Theme themeByMergingTheme:stepTheme withTheme:taskTheme];
     } else if ([element respondsToSelector:@selector(nextResponder)] && [element nextResponder]) {                // continue up responder chain
         id nextResponder = [element nextResponder];
         return [CEVRK1Theme themeForElement:nextResponder];
     } else if ([element respondsToSelector:@selector(parentViewController)] && [element parentViewController]) {  // if has parentViewController, try that route
         UIViewController *parentViewController = [element parentViewController];
         return [CEVRK1Theme themeForElement:parentViewController];
     } else {                                                                                                      // has reached end of chain or not in chain
-        return [CEVRK1Theme defaultTheme];
+        UIViewController *viewController = [[[[[UIApplication sharedApplication] delegate] window] rootViewController] presentedViewController];
```

Comment:
I've switched to a singleton [CEVRK1 fallbackTheme] that gets set every time a StepViewController appears

Reply:
I noticed in recent debugging that the last drawing updates upon the element I was observing (an `ORK1HeadlineLabel`) were triggered by a `moveToWindow` and at that moment, the view was not inside a standard responder chain. In testing a simple survey where there is a style at the survey level and an "override" style in step 2 of 3, in MDH, I'm noticing that it's retaining the style from step 2 onto step 3 because of the fallbackTheme since it cannot check the current step against the task theme and the fallbackTheme is different than the current step.

Here is a test case.
```json
{
  "type": "RKStudioOrderedTask",
  "name": "Survey",
  "identifier": "Survey",
  "styles": {
    "type": "Styles",
    "titleFontSize": 42
  },
  "steps": [
    {
      "type": "QuestionStep",
      "text": "This is text",
      "title": "This is Title",
      "identifier": "Step1",
      "answerFormat": {
        "type": "TextAnswerFormat",
        "isSecureTextEntry": false,
        "multipleLines": false,
        "keyboardType": "Text"
      },
      "optional": false
    },
    {
      "type": "QuestionStep",
      "text": "This is text",
      "title": "This is Title",
      "identifier": "Step1 (Copy)",
      "styles": {
        "type": "Styles",
        "titleColor": "#03befc"
      },
      "answerFormat": {
        "type": "TextAnswerFormat",
        "isSecureTextEntry": false,
        "multipleLines": false,
        "keyboardType": "Text"
      },
      "optional": false
    },
    {
      "type": "QuestionStep",
      "text": "This is text",
      "title": "This is Title",
      "identifier": "Step1 (Copy) (Copy)",
      "answerFormat": {
        "type": "TextAnswerFormat",
        "isSecureTextEntry": false,
        "multipleLines": false,
        "keyboardType": "Text"
      },
      "optional": false
    }
  ]
}
```

Reply:
Dang, good catch. Okay I think I have this working correctly now. I'm seeing the titles correctly colored on each step. And it seems to be working in RKStudioReview. Added a static fallbackTaskViewController property on CEVRK1Theme. If we can't find the theme while moving up the responder chain it will use the fallbackTaskVC's current step.